### PR TITLE
CSI NodeExpansion GA promotion blog

### PR DESCRIPTION
**Pull request by humblec**
KEP#https://github.com/kubernetes/enhancements/issues/3107

### CODE REVIEW

File: content/en/blog/_posts/2023-12-18-node-expand-secret-moves-ga [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+the need to enable the feature gate in control plane components, as was obligatory
+during the alpha phase. As part of this evolution, support has been seamlessly
+integrated into the CSI external-provisioner sidecar controller. To harness this
+feature, ensure your external provisioner sidecar controller is operating at v3.3.0
+or above.
+
+## Navigating Authenticated CSI Storage Resizing
+Assuming all requisite components, including the CSI driver, are deployed and operational
+on your cluster, and you have a CSI driver supporting resizing, you can initiate a
+odeExpansion operation on a CSI volume. Credentials for the CSI NodeExpand operation
```
Comment (sftim) [inline content/en/blog/_posts/2023-12-18-node-expand-secret-moves-ga:48]:
```suggestion
`NodeExpand` operation on a CSI volume. Credentials for the CSI `NodeExpand` operation
```

File: content/en/blog/_posts/2023-12-18-node-expand-secret-moves-ga [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+Upon successful creation of the PersistentVolumeClaim (PVC), you can verify the
+configuration within the .spec.csi field of the PersistentVolume. To confirm,
+execute `kubectl get persistentvolume <pv_name> -o yaml`.
+
+## Engage with the Evolution!
+For those enthusiastic about contributing or delving deeper into the technical
+intricacies, the enhancement proposal comprises exhaustive details about the
+feature's history and implementation. Explore the realms of StorageClass-based
+dynamic provisioning in Kubernetes by referring to the storage class documentation
+and the overarching PersistentVolumes documentation.
```
Comment (sftim) [inline content/en/blog/_posts/2023-12-18-node-expand-secret-moves-ga:89]: I'd make these into 2 hyperlinks.

File: content/en/blog/_posts/2023-12-18-node-expand-secret-moves-ga [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+layout: blog
+title: "Kubernetes 1.29: CSI Storage Resizing Authenticated and Ready for GA in v1.29"
+date: 2023-12-18
+slug: csi-node-expand-secret-support-ga
+---
+**Authors:** Humble Chirammal, Louis Koo (deeproute.ai)
+
+In the dynamic universe of Kubernetes, version v1.29 unfolds a momentous chapter
+as support for authentication during CSI (Container Storage Interface) storage
+resize operations ascends from Beta to General Availability (GA).
```
Comment (sftim) [inline content/en/blog/_posts/2023-12-18-node-expand-secret-moves-ga:11]:
```suggestion
Kubernetes version v1.29 brings generally
available support for authentication during CSI (Container Storage Interface) storage
resize operations.
```

Usually I'm OK with whatever intro folks want, but this didn't ring true. For a lot of clusters this is a niche feature (I personally haven't worked with anyone who used the beta feature); I know it'll be popular with its target audience but “momentous” feels several notches beyond too strong.
So: please reword. I added a suggestion.

File: content/en/blog/_posts/2023-12-18-node-expand-secret-moves-ga [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+Let's embark on the evolution of this feature, initially introduced in alpha in
+Kubernetes v1.25, and unravel the profound changes accompanying its transition to GA.
```
Comment (sftim) [inline content/en/blog/_posts/2023-12-18-node-expand-secret-moves-ga:14]: _profound_? I don't know the truth but I worry that readers will assume an LLM wrote this. Sorry.

**Comment by krol3** [comment]
Hi @humblec ! how are you? Here the Communications Release 1.29 Team, the publication date will be Dec 18th, is that ok for you?

**Comment by humblec** [comment]
> Hi @humblec ! how are you? Here the Communications Release 1.29 Team, the publication date will be Dec 18th, is that ok for you?

Hi @krol3 👋  .. yeah, we are good to publish it on Dec 18th ..

**Comment by humblec** [comment]
@sftim I have addressed the comments

File: content/en/blog/_posts/2023-12-18-node-expand-secret-moves-ga [hunk 5]
```diff
@@ -0,0 +1,92 @@
+---
+layout: blog
+title: "Kubernetes 1.29: CSI Storage Resizing Authenticated and Generally Available in v1.29"
+date: 2023-12-18
```
Comment (krol3) [inline content/en/blog/_posts/2023-12-18-node-expand-secret-moves-ga:4]: @humblec  Please be ready to be published on Dec 11th

Comment (krol3) [inline content/en/blog/_posts/2023-12-18-node-expand-secret-moves-ga:4]:
```suggestion
date: 2023-12-11
```

**Comment by humblec** [comment]
@krol3 I have adjusted the date as per our discussion .. ptal. thanks.

File: content/en/blog/_posts/2023-12-11-node-expand-secret-moves-ga
Comment (sftim) [inline content/en/blog/_posts/2023-12-11-node-expand-secret-moves-ga:1]: Hi. The filename should be `content/en/blog/_posts/2023-12-11-node-expand-secret-moves-ga.md`

File: content/en/blog/_posts/2023-12-11-node-expand-secret-moves-ga [hunk 6]
```diff
@@ -0,0 +1,92 @@
+---
+layout: blog
+title: "Kubernetes 1.29: CSI Storage Resizing Authenticated and Generally Available in v1.29"
+date: 2023-12-11
+slug: csi-node-expand-secret-support-ga
+---
+**Authors:** Humble Chirammal, Louis Koo (deeproute.ai)
```
Comment (sftim) [inline content/en/blog/_posts/2023-12-11-node-expand-secret-moves-ga:7]: Did you mean:
```markdown
**Authors:** Humble Chirammal (deeproute.ai), Louis Koo (deeproute.ai)
```
?

**Review by sftim — DISMISSED** [review]
Thanks.

- Please fix the filename per inline feedback.
- Please change the base (target) branch to _main_. Even though this is related to v1.29, it is not part of the v1.29 documentation and will merge in separately.

File: content/en/blog/_posts/2023-12-11-node-expand-secret-moves-ga (hunk 6 above)
Comment (humblec) [reply content/en/blog/_posts/2023-12-11-node-expand-secret-moves-ga]: no.. :). I have added the company name too.

File: content/en/blog/_posts/2023-12-11-node-expand-secret-moves-ga
Comment (humblec) [reply content/en/blog/_posts/2023-12-11-node-expand-secret-moves-ga:1]: Done

**Comment by humblec** [comment]
> Thanks.
> 
> * Please fix the filename per inline feedback.
> * Please change the base (target) branch to _main_. Even though this is related to v1.29, it is not part of the v1.29 documentation and will merge in separately.

Addressed both. Thanks @sftim !

**Comment by sftim** [comment]
Assuming that the date from https://github.com/kubernetes/website/pull/43921#discussion_r1395772303 is the one to use.

**Comment by sftim** [comment]
/hold
Do not merge until Kubernetes v1.29 is released.

/lgtm
/approve

**Comment by krol3** [comment]
Hi @humblec,  Release 1.29 was posponed, Please assign new  date on Dec 15th. cc: @Priyankasaggu11929 @sftim

**Comment by humblec** [comment]
@krol3 and @sftim as suggested I have changed the date to 2012-12-15 . Ptal .. thanks 👍

File: content/en/blog/_posts/2023-12-15-node-expand-secret-moves-ga.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+a passphrase is mandated to expand the device and grow the filesystem, underscoring
+the necessity for authenticated resizing.
+
+## What's new for Kubernetes v1.29
+With the graduation to GA, the feature remains enabled by default. Support for
+node-level volume expansion secrets has been seamlessly integrated into the CSI
+external-provisioner sidecar controller. To take advantage, ensure your external
+CSI storage provisioner sidecar controller is operating at v3.3.0 or above.
+
+## Navigating Authenticated CSI Storage Resizing
```
Comment (sftim) [inline content/en/blog/_posts/2023-12-15-node-expand-secret-moves-ga.md:41]:
```suggestion
## Navigating authenticated CSI storage resizing
```

**Comment by sftim** [comment]
/lgtm
/approve

Pending feedback is nice to have and does not block publication.

:stop_sign: Do not merge until Kubernetes 1.29 is released. :stop_sign:

**Pull request merged**